### PR TITLE
chore: add condition for only building on main

### PR DESCRIPTION
**Pull request by BobyMCbobs**
ensure that longer-to-build images only get built on main

### CODE REVIEW

File: config.yaml [hunk 1]
```diff
@@ -86,7 +86,10 @@ build:
     destination: ghcr.io/geonet/base-images/chart-centos7:latest
   - source: ./images/rpmbuild-centos7/Dockerfile
     destination: ghcr.io/geonet/base-images/rpmbuild-centos7:latest
+    buildOnMain: true
```
Comment (Mossman1215) [inline config.yaml:89]: is there a suitable way to highlight that this is a restriction to only build on main
vs buildOnAll
onlyBuildOnMain?
buildOnMainOnly?

Comment (BobyMCbobs) [reply config.yaml:89]: Yeah thought about it for a few minutes. Good for feedback.

How about _requireMainBranch_?

Comment (BobyMCbobs) [reply config.yaml]: Went with _buildOnMainOnly_

**Review by Mossman1215 — APPROVED** [review]
thanks!

**Pull request merged**